test/functional/lua/commands_spec.lua

```diff
@@ -19,6 +19,88 @@ local curbufmeths = helpers.curbufmeths
 
 before_each(clear)
 
+local empty_screen =
+[[
+      ^                                                  |
+      {1:~                                                 }|
+      {1:~                                                 }|
+      {1:~                                                 }|
+      {1:~                                                 }|
+      {1:~                                                 }|
+      {1:~                                                 }|
+      {1:~                                                 }|
+      {1:~                                                 }|
+                                                        |
+]]
+
+local disptest = function(args)
+  describe(args.descrip, function()
+  it(args.descrip, function()
+  local screen = Screen.new(50,10)
+  screen:attach()
+  screen:set_default_attr_ids({
+    [1] = {bold = true, foreground = Screen.colors.Blue1},
+    [2] = {bold = true, reverse = true},
+    [3] = {foreground = Screen.colors.Grey100, background = Screen.colors.Red},
+    [4] = {bold = true, foreground = Screen.colors.SeaGreen4},
+  })
+
+  feed(args.cmd)
+  screen:expect(args.expected_scr)
+  feed('<cr>')
+  screen:expect(empty_screen)
+  eq(args.expected_msg, eval('v:errmsg'))
+
+  local status, err = pcall(command,'lua error("some error\\nin a\\nAPI command")')
+  local expected = 'Vim(lua):E5105: Error while calling lua chunk: [string "<VimL compiled string>"]:1: some error\nin a\nAPI command'
+  eq(false, status)
+  eq(expected, string.sub(err, -string.len(expected)))
+
+  feed(':messages<cr>')
+  screen:expect(args.expected_scr)
+    end)
+  end)
+end
+
+disptest{ descrip = 'can show multiline error messages',
+          cmd     = ':lua error("fail\\nmuch error\\nsuch details")<cr>',
+	  expected_msg = 'E5105: Error while calling lua chunk: [string ' ..
+	                 '"<VimL compiled string>"]:1: fail\nmuch error\n' ..
+			 'such details',
+	  expected_scr = [[
```

Comment:
spaces for indent, not tabs.

Avoid "DRY" in this early stage. Often it's more readable just to put multiple tests in a single `it()` instead of parameterizing the hell out of the test and using a function.

Factoring out some common code in functions is good to avoid lots of redundancy. But some redundancy is ok. Avoid going nuts with "DRY".